### PR TITLE
Add feed validation to Follow requests from Web App and API to avoid invalid feeds

### LINKED ISSUE #172: Validate New Follow requests
**Is your feature request related to a problem? Please describe.**

As a user you can currently "Follow" (_by hand_) any new User or Feed by their Twtxt URI.

- Click "Follow":

Screen Shot 2020-09-07 at 11 00 03

- Fill our the form:

Screen Shot 2020-09-07 at 11 00 07

**Describe the solution you'd like**

What is missing is the following (_a lot of code exists in the codebase to help with this_):

- Validate that new User/Feed being entered is _actually_ valid.

As an _optional_ bonus (_for any senior contributors_) would be also accept an RSS/Atom feed and implicitly add this to the [feeds.twtxt.net](https://feeds.twtxt.net) service (_backend API does not currently exists to support this_) so that is is trivial to add both Twtxt and RSS/Atom feeds. Making it "just" that little bit easier.

**Describe alternatives you've considered**

N/A

**Additional context**

N/A

### PR DESCRIPTION
**Pull request by prologic**
##### Summary

ssia

##### Component Name

- area/backend

##### Test Plan

- Tested locally
- Imported my feeds from twtxt.net
- Make sure I can still follow stuff
- Meka sure trying to follow a bad feed fails.

##### Additional Information

Fixes #172

### CODE REVIEW

File: internal/follow_handlers.go [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+
+		ctx.Error = false
+		ctx.Message = fmt.Sprintf("Successfully started following %s: %s", nick, url)
+		s.render("error", w, ctx)
+		return
+	}
+}
+
+// ImportHandler ...
+func (s *Server) ImportHandler() httprouter.Handle {
```
Comment (dooven) [inline internal/follow_handlers.go:140]: oh wow this is neat

File: internal/follow_handlers.go [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+			url = UserURL(url)
+			nick := NormalizeUsername(filepath.Base(url))
+			followee, err := s.db.GetUser(nick)
+			if err != nil {
+				log.WithError(err).Warnf("error loading user object for followee %s", nick)
+			} else {
+				if followee.Followers != nil {
+					delete(followee.Followers, user.Username)
+					if err := s.db.SetUser(followee.Username, followee); err != nil {
+						log.WithError(err).Warnf("error updating user object for followee %s", followee.Username)
```
Comment (dooven) [inline internal/follow_handlers.go:248]: should we return an error page on this error case?

File: internal/follow_handlers.go (hunk 1 above)
Comment (prologic) [reply internal/follow_handlers.go:140]: Yeah it doesn't have any validation though! Its a way to "bulk import" your "following" list from another Twtxt client 👍

File: internal/follow_handlers.go (hunk 2 above)
Comment (prologic) [reply internal/follow_handlers.go:248]: Nah not here. I want to refactor all of this with #234 234 with an event-sourcing / pubsub pattern instead. All of this "special" `twtxt` bot handling code is a mess :/

**Comment by prologic** [comment]
Rebased

**Review by dooven — APPROVED** [review]

**Pull request merged**